bgpd/bgp_fsm.c

```diff
@@ -2055,3 +2256,498 @@ int bgp_event_update(struct peer *peer, int event)
 
 	return ret;
 }
+/* BGP GR Code */
+
+int bgp_gr_lookup_n_update_all_peer(struct bgp *bgp,
+			enum global_mode global_new_state,
+			enum global_mode global_old_state)
+{
+	struct peer *peer = {0};
+	struct listnode *node = {0};
+	struct listnode *nnode = {0};
+	enum peer_mode  peer_old_state = PEER_INVALID;
+
+	for (ALL_LIST_ELEMENTS(bgp->peer, node, nnode, peer)) {
+
+		if (BGP_DEBUG(graceful_restart, GRACEFUL_RESTART))
+			zlog_debug(
+			"%s BGP_GR:: Peer: (%s) :",
+				__func__, peer->host);
+
+		peer_old_state = bgp_peer_gr_mode_get(peer);
+
+		if (peer_old_state == PEER_GLOBAL_INHERIT) {
+
+		/*
+		 *Reset only these peers and send a
+		 *new open message with the change capabilities.
+		 *Considering the mode to be "global_new_state" and
+		 *do all operation accordingly
+		 */
+
+			switch (global_new_state) {
+			case GLOBAL_HELPER:
+				BGP_PEER_GR_HELPER_ENABLE(peer);
+				break;
+			case GLOBAL_GR:
+				BGP_PEER_GR_ENABLE(peer);
+				break;
+			case GLOBAL_DISABLE:
+				BGP_PEER_GR_DISABLE(peer);
+				break;
+			case GLOBAL_INVALID:
+				zlog_debug(
+				"%s BGP_GR::GLOBAL_INVALID",
```

Comment:
shoudl be "%s: ..."

Reply:
> shoudl be "%s: ..."

In all the places where i am printing function name before "BGP_GR", i am leaving a space i.e " " instead of :, so that would be fine